src/main/java/teammates/ui/controller/FeedbackResendLinksAction.java

```diff
@@ -0,0 +1,97 @@
+package teammates.ui.controller;
+
+import teammates.common.exception.EmailSendingException;
+import teammates.common.exception.EntityDoesNotExistException;
+import teammates.common.exception.EntityNotFoundException;
+import teammates.common.exception.TeammatesException;
+import teammates.common.util.Const;
+import teammates.common.util.EmailWrapper;
+import teammates.common.util.FieldValidator;
+import teammates.common.util.Logger;
+import teammates.common.util.StatusMessage;
+import teammates.common.util.StatusMessageColor;
+import teammates.logic.api.EmailGenerator;
+import teammates.ui.pagedata.FeedbackResendLinksPageData;
+import teammates.ui.pagedata.PageData;
+
+public class FeedbackResendLinksAction extends Action {
+
+    private static final Logger log = Logger.getLogger();
+
+    @Override
+    protected void authenticateUser() {
+        // This feature does not require authentication
+    }
+
+    @Override
+    protected ActionResult execute() throws EntityDoesNotExistException {
+        try {
+            FieldValidator validator = new FieldValidator();
+            String userEmailToResend = getNonNullRequestParamValue(Const.ParamsNames.STUDENT_EMAIL);
+
+            statusToAdmin = "Resend links requested"
+                    + "<br>Email: " + userEmailToResend;
+            String error = validator.getInvalidityInfoForEmail(userEmailToResend);
+
+            if (!error.isEmpty()) {
+                statusToUser.add(new StatusMessage(error, StatusMessageColor.DANGER));
+                PageData data = new FeedbackResendLinksPageData(account, sessionToken, error);
+                return createAjaxResult(data);
+            }
+
+            boolean shouldSkipVerification = false;
+            if (gateKeeper.getCurrentUser() != null) {
```

Comment:
Can help me understand when this is null?

Reply:
Since the user does not need to log in to use this app, therefore when the user does not log in, the variable will be null.
![image](https://user-images.githubusercontent.com/28507005/39673922-1ebffbe6-5177-11e8-874f-7cc17141b27e.png)